internal/ingress/annotations/cors/main.go

```diff
@@ -123,16 +123,18 @@ func (c cors) Parse(ing *networking.Ingress) (interface{}, error) {
 
 	unparsedOrigins, err := parser.GetStringAnnotation("cors-allow-origin", ing)
 	if err == nil {
-		config.CorsAllowOrigin = strings.Split(unparsedOrigins, ",")
-		for i, origin := range config.CorsAllowOrigin {
+		origins := strings.Split(unparsedOrigins, ",")
+		for _, origin := range origins {
 			origin = strings.TrimSpace(origin)
 			if origin == "*" {
 				config.CorsAllowOrigin = []string{"*"}
 				break
 			}
-			if !corsOriginRegex.MatchString(origin) {
+
+			if corsOriginRegex.MatchString(origin) {
+				config.CorsAllowOrigin = append(config.CorsAllowOrigin, origin)
+			} else {
```

Comment:
Good catch on this!

Just a nit suggestion to simplify the code: instead of going with the if and else here, do something like "if bad condition then log error and continue interaction to the next item"

This way you can early fail, and save one "else" and indentation here. 

Also, I'm not sure we are initializing the config.CorsAllowOrigins somewhere (sorry, on mobile here) so let's make sure we initialize this array before using/appending (again sorry if we do but I'm away from computer)

Reply:
Added if-else rewrite, along with initializing the CorsAllowOrigin, thanks for the note!